lib/core/include/apiHandler.hpp

```diff
@@ -97,235 +100,241 @@ namespace irods {
-                std::string    in_pack_key;
-                std::string    out_pack_key;
-                std::string    in_pack_value;
-                std::string    out_pack_value;
-                std::string    operation_name;
-
-                lookup_table< std::string>   extra_pack_struct;
-
-                std::function<void( void* )> clearInStruct;		//free input struct function
-
-        private:
-            template<typename... types_t>
-            error invoke_policy_enforcement_point(
-                rule_engine_context_manager<
-                                    unit,
-                                    ruleExecInfo_t*,
-                                    AUDIT_RULE > _re_ctx_mgr,
-                plugin_context      _ctx,
-                const std::string&  _operation_name,
-                const std::string&  _class,
-                types_t...          _t) {
-                bool ret = false;
-                error saved_op_err = SUCCESS();
-                for ( auto& ns : NamespacesHelper::Instance()->getNamespaces() ) {
-                    std::string rule_name = ns + "pep_" + _operation_name + "_" + _class;
-
-                    if (RuleExistsHelper::Instance()->checkOperation( rule_name ) ) {
-                        if (_re_ctx_mgr.rule_exists(rule_name, ret).ok() && ret) {
-                            error op_err = _re_ctx_mgr.exec_rule(
-                                               rule_name,
-                                               instance_name_,
-                                               _ctx,
-                                               std::forward<types_t>(_t)...);
-                            if (!op_err.ok()) {
-                                rodsLog(
-                                     LOG_DEBUG,
-                                    "%s-pep rule [%s] failed with error code [%d]",
-                                    _class.c_str(),
-                                    rule_name.c_str(),
-                                    op_err.code());
-                                saved_op_err = op_err;
-                            }
-                        } else {
-                            rodsLog(
-                                LOG_DEBUG10,
-                                "Rule [%s] passes regex test, but does not exist",
-                                rule_name.c_str() );
+                    return op_err.code();
+                }
+
+                return op_err.code();
+#else // ENABLE_RE
+                return fcn(_comm, _t...);
+#endif // ENABLE_RE
+            }
+            catch (const boost::bad_any_cast&) {
+                std::string msg( "failed for call - " );
+                msg += operation_name;
+                irods::log(ERROR(INVALID_ANY_CAST, msg));
+                return INVALID_ANY_CAST;
+            }
+
+            return 0;
+        } // call_handler
+
+        // =-=-=-=-=-=-=-
+        // ctors
+        api_entry( apidef_t& );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'api_entry' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5139965)